core/chain_manager_test.go

```diff
@@ -483,22 +485,71 @@ func TestInsertNonceError(t *testing.T) {
 	}
 }
 
-/*
-func TestGenesisMismatch(t *testing.T) {
-	db, _ := ethdb.NewMemDatabase()
-	var mux event.TypeMux
-	genesis := GenesisBlock(0, db)
-	_, err := NewChainManager(genesis, db, db, db, thePow(), &mux)
-	if err != nil {
+func TestChainReorgMissingTransactions(t *testing.T) {
```

Comment:
This test is wrong. The two transactions created in the two forks are not the same. Signing introduces a randomness into the signature, causing the two transaction to have different signatures, hence different hashes. If you were to define one single transaction

```
	txn, _ := types.NewTransaction(0, addr2, big.NewInt(1000), params.TxGas, nil, nil).SignECDSA(key1)
```

and inject that into both chains, your tests will fail, per the issue found by @bas-vk .